manticore/core/executor.py

```diff
@@ -760,7 +760,7 @@ def setstate(state, val):
                     with current_state as new_state:
                         children.append(new_state.co)
                         new_state.add(e.constraint)
-                        self.newerror(new_state.current.PC)
+                        self.newerror(new_state.model.current.PC)
```

Comment:
Will do! That is cleaner. Why do we use the the with-statement to make forked states, instead of a fork method or something?